src/server.rs

```diff
@@ -56,6 +57,7 @@ pub struct Server {
     directories: Arc<RwLock<Vec<Directory>>>,
     startup_handler: Option<Arc<FunctionInfo>>,
     shutdown_handler: Option<Arc<FunctionInfo>>,
+    memory_store: Arc<Mutex<HashMap<String, Vec<u32>>>>,
```

Comment:
Do we need this to be wrapped in an Arc Mutex? This comes with some overhead, since this method always come from python, we can assume only one thread will be reading / writing from it at a time. I'm not sure if we can get a `&mut self` in the `get_calls_list` method

Reply:
If you think we will be accessing this concurrently or can't get a `&mut self` reference we should use `Arc<Dashmap<...>>` this is already imported and would do better if we do hit this at the same time

Reply:
Also another query, will we need a background thread to clear these out? Current if a client never requests again the entry will stay indefinitely, I could see memory growing if used by a lot of people